pkg/operator/operator/validations.go

```diff
@@ -744,3 +786,53 @@ func getValidationK8sResources() ([]kunstructured.Unstructured, *kresource.Quant
 
 	return virtualServices, maxMem, err
 }
+
+func validateOverridenImages(images, keys []string) error {
```

Comment:
I think this can be simplified a bit, although I'm trying to wrap my head around exactly how. I've typed out two ideas, realized there is a good reason you designed it this way, and this is now my 3rd version of my proposal, which is more modest of a change than my first two ideas :) :

Perhaps just a single loop over images/keys would be more clear. We could declare a pointer variable for `ecrAuth` outside of the loop, and if we end up needing it in the loop, we can check if it's still nil, and if it is, then generate it. Making the code to generate it into a helper function may also help with clarity. What do you think?